### ISSUE TITLE
Dataset: rust-ai Ecosystem - Training Data from tzervas Crates

### ISSUE THREAD

**Issue by tzervas** [tzervas/tritter#83]
## Overview

Extract high-quality training data from the rust-ai ecosystem crates to ensure Tritter has deep knowledge of its own acceleration framework.

## Target Crates (github.com/tzervas)

| Crate | Version | Focus Area |
|-------|---------|------------|
| aphelion-core | 1.2.6 | Pipeline composition, backend abstraction |
| aphelion-macros | 1.2.6 | Model builder procedural macros |
| axolotl-rs | 1.1.1 | Fine-tuning toolkit patterns |
| bitnet-quantize | 0.1.1 | BitNet 1.58-bit implementation |
| peft-rs | 1.0.1 | 11 PEFT adapter types |
| qlora-rs | 1.0.1 | NF4 quantization, PagedAdamW |
| rust-ai-core | 0.2.7 | Device selection, config validation |
| trit-vsa | 0.1.1 | Ternary arithmetic, VSA operations |
| tritter-accel | 0.1.1 | Tritter-specific acceleration |
| unsloth-rs | 1.0.1 | Fused attention, RoPE, RMSNorm |
| vsa-optim-rs | 0.1.1 | Gradient compression, phase training |

### 1. Source Code
- Clone all repos from GitHub
- Extract Rust source files
- Preserve module structure and documentation comments

### 2. docs.rs Documentation
- Scrape API documentation for each crate
- Extract module docs, struct docs, function docs
- Preserve code examples

### 3. Synthetic Q&A Generation
- Generate question-answer pairs about API usage
- Create "how to use X" instruction examples
- Generate troubleshooting scenarios

## Tasks

- [ ] Create scraper script: \`scripts/data/scrape_rustai_ecosystem.py\`
- [ ] Clone all 11 tzervas repos
- [ ] Extract source code with inline docs
- [ ] Scrape docs.rs pages for each crate version
- [ ] Parse API documentation into structured format
- [ ] Generate synthetic Q&A pairs (using existing LLM)
- [ ] Create code completion examples
- [ ] Target output: \`data/instruct/rust-ai-ecosystem/\`

## Output Format

\`\`\`json
{
  "source": "rust-ai-ecosystem",
  "crate": "bitnet-quantize",
  "type": "api_doc",
  "module": "quantization",
  "item": "BitLinear",
  "doc": "A drop-in replacement for linear layers...",
  "code_example": "let layer = BitLinear::from_weight(&weight, None, &config)?;",
  "usage_qa": [
    {"q": "How do I quantize weights with BitNet?", "a": "Use BitLinear::from_weight..."}
  ]
}
\`\`\`

## Priority

This is **essential** for ensuring Tritter can effectively use and reason about its own acceleration tooling.

**Parent issue**: #75
